compiler/util/stringutil.cpp

```diff
@@ -435,16 +435,31 @@ std::string ltrimAllLines(std::string s) {
 }
 
 /*
- * Gather words from the string and store them into the array.
- * These words are arguments to a program.
+ * Split a string separated by a delimiter into a vector of substrings.
+ * If splitAllWhitespaces is true, split the string against all whitespace characters.
  */
-void readArgsFromString(std::string s, std::vector<std::string>& args) {
-  if (s != "") {
-    //split s by spaces
-    std::stringstream argsStream(s);
-    std::string word;
-    while(argsStream >> word)
-      args.push_back(word);
+void splitString(const std::string& s, std::vector<std::string>& vec, char delim, bool splitAllWhitespaces) {
```

Comment:
Hmm... If I call this function with  `delim=','` then I wouldn't expect it to split on whitespace.
How about this - since the whitespace version uses a list of token characters - why not just do that? And if you want to split on `,` and `\n` then you'd call it with `",\n"` ?